stdlib/REPL/src/LineEdit.jl

```diff
@@ -1272,8 +1272,30 @@ function write_prompt(terminal, p::Prompt)
     return width
 end
 
+# On Windows, when launching external processes, we cannot control what assumption they make on the
+# console mode. We thus forcibly reset the console mode at the start of the prompt to ensure they do
+# not leave the console mode in a corrupt state.
+if Sys.iswindows()
+function _console_mode()
+    hOutput = ccall(:GetStdHandle, stdcall, Int32, (Int32,), -11) # STD_OUTPUT_HANDLE
+    dwMode = Ref{UInt32}(0)
+    ccall(:GetConsoleMode, stdcall, Int32, (Int32, Ptr{Nothing}), hOutput, dwMode)
+    dwMode[]
+end
+const default_console_mode = _console_mode()
+function _reset_console_mode()
+    mode = _console_mode()
+    if mode !== default_console_mode
+        hOutput = ccall(:GetStdHandle, stdcall, Int32, (Int32,), -11) # STD_OUTPUT_HANDLE
```

Comment:
it's a OS_HANDLE, not an Int32

Reply:
Fixed. Also updated to be more pedantic about DWORD being a a UInt32 and avoiding the implicit cast using an `unsafe_trunc`